orchagent/routeorch.cpp

```diff
@@ -1038,7 +1037,19 @@ bool RouteOrch::addNextHopGroup(const NextHopGroupKey &nexthops)
      * and add the corresponding next_hop_id to next_hop_ids. */
     for (auto it : next_hop_set)
     {
-        if (!m_neighOrch->hasNextHop(it))
+        sai_object_id_t next_hop_id;
+        if (m_neighOrch->hasNextHop(it))
+        {
+            next_hop_id = m_neighOrch->getNextHopId(it);
+        }
+        /* See if there is an IP neighbor NH for MPLS NH*/
+        else if (it.isMplsNextHop() &&
```

Comment:
do not follow this elseif logic. What is the intention here ?
If the `m_neighOrch->hasNextHop(it)` is true how it will even reach elseif logic.

Reply:
@abdosi The logic is this:  
1) check to see if NextHop (IP or MPLS) exists.  If it does, use existing NH OID.
2) If the NextHop does not exist and is MPLS, check if Neighbor has been resolved (check for matching IP-only NH).  If MPLS NH does not exist but matching IP-only NH exists, then Neighbor is resolved and we must explicitly add MPLS NH.
3) If Neighbor has not been resolved (matching IP-only NH does not exist), then return failed.

Reply:
so just to clarify these cover the scenario n-1:1 mapping of nexthop to neighbor that mpls brings where neighbors is already created but 2nd nexthop onwards are not created as part of `NeighOrch::addNeighbor()`

Reply:
@abdosi Exactly

Reply:
@abdosi Per discussion with @prsunny, this logic has been reverted and new ::removeMplsNextHop is added.